lib/services/local/resource.go

```diff
@@ -97,6 +98,15 @@ func itemsFromResource(resource types.Resource) ([]backend.Item, error) {
 		item, err = itemFromClusterNetworkingConfig(r)
 	case types.AuthPreference:
 		item, err = itemFromAuthPreference(r)
+	case types.Resource153Unwrapper:
+		switch r153 := r.Unwrap().(type) {
+		case *autoupdatev1pb.AutoUpdateConfig:
+			item, err = itemFromAutoUpdateConfig(r153)
+		case *autoupdatev1pb.AutoUpdateVersion:
+			item, err = itemFromAutoUpdateVersion(r153)
+		default:
+			return nil, trace.NotImplemented("cannot itemFrom resource of type %T", resource)
+		}
```

Comment:
Instead of doing a nested switch we can do a single switch like in `applyResources`

```code
// ItemsFromResource attempts to construct one or more instances of `backend.Item` from
// a given resource.  If `rsc` is not one of the supported resource types,
// a `trace.NotImplementedError` is returned.
func itemsFromResource(resource types.Resource) ([]backend.Item, error) {
	var item *backend.Item
	var extItems []backend.Item
	var err error

	// Unwrap "new style" resources.
	// We always want to switch over the underlying type.
	var res any = resource
	if w, ok := res.(types.Resource153Unwrapper); ok {
		res = w.Unwrap()
	}

	switch r := res.(type) {
```